src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -22872,5 +22871,174 @@ void M<T>()
+        try
+        {
+            var u = Unchecked((-1, -1));
+            Console.WriteLine(u);
+            var c = Checked((-1, -1));
+            Console.WriteLine(c);
+        }
+        catch (OverflowException)
+        {
+            Console.WriteLine(""overflow"");
+        }
+    }
+}";
+            var comp = CreateStandardCompilation(
+                source,
+                references: new[] { ValueTupleRef, SystemRuntimeFacadeRef },
+                options: TestOptions.ReleaseExe);
+            var verifier = CompileAndVerify(comp, expectedOutput:
+@"(-1, 255)
+overflow");
+            verifier.VerifyIL("C.Unchecked",
+@"{
+  // Code size       22 (0x16)
+  .maxstack  2
+  .locals init (System.ValueTuple<int, int> V_0)
+  IL_0000:  ldarg.0
+  IL_0001:  stloc.0
+  IL_0002:  ldloc.0
+  IL_0003:  ldfld      ""int System.ValueTuple<int, int>.Item1""
+  IL_0008:  conv.i8
+  IL_0009:  ldloc.0
+  IL_000a:  ldfld      ""int System.ValueTuple<int, int>.Item2""
+  IL_000f:  conv.u1
+  IL_0010:  newobj     ""System.ValueTuple<long, byte>..ctor(long, byte)""
+  IL_0015:  ret
+}");
+            verifier.VerifyIL("C.Checked",
+@"{
+  // Code size       22 (0x16)
+  .maxstack  2
+  .locals init (System.ValueTuple<int, int> V_0)
+  IL_0000:  ldarg.0
+  IL_0001:  stloc.0
+  IL_0002:  ldloc.0
+  IL_0003:  ldfld      ""int System.ValueTuple<int, int>.Item1""
+  IL_0008:  conv.i8
+  IL_0009:  ldloc.0
+  IL_000a:  ldfld      ""int System.ValueTuple<int, int>.Item2""
+  IL_000f:  conv.ovf.u1
+  IL_0010:  newobj     ""System.ValueTuple<long, byte>..ctor(long, byte)""
+  IL_0015:  ret
+}");
+        }
+
+        [Fact(Skip = "19398")]
+        [WorkItem(18459, "https://github.com/dotnet/roslyn/issues/18459")]
+        [WorkItem(19398, "https://github.com/dotnet/roslyn/issues/19398")]
+        public void CheckedDeconstructionConversions()
+        {
+            var source =
+@"using System;
+class C
+{
+    static void Unchecked((int, int) t, out long a, out byte b)
+    {
+        unchecked
+        {
+            (a, b) = ((long, byte))t;
+        }
+    }
+    static void Checked((int, int) t, out long a, out byte b)
+    {
+        checked
+        {
+            (a, b) = ((long, byte))t;
+        }
+    }
+    static void Main()
+    {
+        try
+        {
+            long a;
+            byte b;
+            Unchecked((-1, -1), out a, out b);
+            Console.WriteLine((a, b));
+            Checked((-1, -1), out a, out b);
+            Console.WriteLine((a, b));
+        }
+        catch (OverflowException)
+        {
+            Console.WriteLine(""overflow"");
+        }
+    }
+}";
+            var comp = CreateStandardCompilation(
+                source,
+                references: new[] { ValueTupleRef, SystemRuntimeFacadeRef },
+                options: TestOptions.ReleaseExe);
+            var verifier = CompileAndVerify(comp, expectedOutput:
+@"(-1, 255)
+overflow");
+            verifier.VerifyIL("C.Unchecked",
+@"{
+ ...
```

Comment:
Oh, nevermind, this is skipped ;-)

---
In reply to: [116064973](https://github.com/dotnet/roslyn/pull/19418#discussion_r116064973) [](ancestors = 116064973)